packages/frontend/src/components/ExportCSV/index.tsx

```diff
@@ -50,36 +51,36 @@ const ExportCSV: FC<ExportCSVProps> = memo(
         const [limit, setLimit] = useState<string>(Limit.TABLE);
         const [customLimit, setCustomLimit] = useState<number>(1);
         const [format, setFormat] = useState<string>(Values.FORMATTED);
-        const [isExporting, setIsExporting] = useState<boolean>(false);
+
+        const { isFetching: isExporting, refetch: handleExport } = useQuery(
+            [limit, customLimit, rows, format],
+            () =>
+                getCsvLink(
+                    limit === Limit.CUSTOM
+                        ? customLimit
+                        : limit === Limit.TABLE
+                        ? rows?.length ?? 0
+                        : null,
+                    format === Values.RAW,
+                )
+                    .then((url) => {
+                        if (url) window.open(url, '_blank');
+                    })
+                    .catch((error) => {
+                        showToastError({
+                            title: `Unable to download CSV`,
+                            subtitle: error?.error?.message,
+                        });
+                    }),
```

Comment:
can you move this to onSuccess and onError blocks. those are react-query options that you you can write in an options object (where you have `enabled: false`)